File: 08_binary_search/08_09_find_minimum_in_rotated_sorted_array/find_minimum_in_rotated_sorted_array.js
function findMin(nums) {
    let low = 0;
    let high = nums.length - 1;
    let result = Infinity;
    while (low <= high) {
        let mid = low + Math.floor((high - low) / 2);
        if (nums[low] <= nums[mid]) {
            result = Math.min(result, nums[low]);
            low = mid + 1;
        } else {
            result = Math.min(result, nums[mid]);
            high = mid - 1;
        }
    }
    return result;
}

// Test cases
let nums1 = [3, 4, 5, 1, 2];
console.log(`Test Case 1 - Array: [${nums1}]`);
console.log(`Minimum Element: ${findMin(nums1)}`);

let nums2 = [4, 5, 6, 7, 0, 1, 2];
console.log(`\nTest Case 2 - Array: [${nums2}]`);
console.log(`Minimum Element: ${findMin(nums2)}`);

let nums3 = [11, 13, 15, 17];
console.log(`\nTest Case 3 - Array: [${nums3}]`);
console.log(`Minimum Element: ${findMin(nums3)}`);

let nums4 = [5, 5, 5, 5, 5];
console.log(`\nTest Case 4 - Array: [${nums4}]`);
console.log(`Minimum Element: ${findMin(nums4)}`);

File: 08_binary_search/08_09_find_minimum_in_rotated_sorted_array/find_minimum_in_rotated_sorted_array.py
def find_min(nums):
    low, high = 0, len(nums) - 1
    result = float('inf')
    while low <= high:
        mid = low + (high - low) // 2
        if nums[low] <= nums[mid]:
            result = min(result, nums[low])
            low = mid + 1
        else:
            result = min(result, nums[mid])
            high = mid - 1
    return result

# Test cases
nums1 = [3, 4, 5, 1, 2]
print(f"Test Case 1 - Array: {nums1}")
print(f"Minimum Element: {find_min(nums1)}")

nums2 = [4, 5, 6, 7, 0, 1, 2]
print(f"\nTest Case 2 - Array: {nums2}")
print(f"Minimum Element: {find_min(nums2)}")

nums3 = [11, 13, 15, 17]
print(f"\nTest Case 3 - Array: {nums3}")
print(f"Minimum Element: {find_min(nums3)}")

nums4 = [5, 5, 5, 5, 5]
print(f"\nTest Case 4 - Array: {nums4}")
print(f"Minimum Element: {find_min(nums4)}")
